build-container-image.sh

```diff
@@ -0,0 +1,43 @@
+#!/bin/bash
+
+docker_force=false
+while getopts D opt; do
+    case $opt in
+        D) docker_force=true ;;
+        ?) echo "script usage: $(basename $0) [-D]" >&2; exit 1;;
+    esac
+done
+
+# remove the parsed options from the positional params
+shift $(( OPTIND - 1 ))
+container_tag_name=$1
+
+if $docker_force; then
+    if ! command -v docker &> /dev/null; then
+              echo "Docker is not installed. Please install Docker first."
+              exit 1
+          fi
+
+          if ! docker info &> /dev/null; then
+              echo "Docker daemon is not running. Please start the Docker daemon."
+              exit 1
+          fi
+
+          if ! docker ps &> /dev/null; then
+              echo "You don't have access to the Docker daemon without sudo."
+              echo "Please add your user to the 'docker' group or run this script with sudo."
+              exit 1
+          fi
+          docker build -t "$container_tag_name" .
+else
+    if ! command -v podman &> /dev/null; then
```

Comment:
NIT: Moving the stuff into functions, could make the check for a command reusable. Having a global function like `(COMMAND) -> Exit Code` would be quite convenient IMHO